### PR TITLE
Remove @storybook/addon-knobs from the repository

### PR DESCRIPTION
**Pull request by ciampo**
## What?

Remove the `@storybook/addon-knobs` dependency from the repository

## Why?

The dependency has been deprecated for a long time, and we recently removed all usages from all Storybook examples.

This dependency is currently blocking using npm 8 (see https://github.com/WordPress/gutenberg/issues/46443)

## How?

Removed from:
- npm dependency
- docs
- Storybook settings

## Testing Instructions

- Make sure that the project builds correctly
- Make sure that Storybook examples load and function as on `trunk`
- Do a smoke test on the editor, make sure that everything works as on `trunk`

### CODE REVIEW

**Comment by ciampo** [comment]
Errors are expected at this point, until #47149 and #47117 are merged. I will then rebase and mark this PR as ready

**Comment by tyxla** [comment]
I'm refraining from a review while this is a draft, but please feel free to re-request one once this is ready, @ciampo.

**Comment by ciampo** [comment]
All knobs usages should have been migrated. Time to give this PR a review!

File: packages/components/CONTRIBUTING.md [hunk 1]
```diff
@@ -361,7 +361,7 @@ Primary.args = {
 };
 ```
 
-A great tool to use when writing stories is the [Storybook Controls addon](https://storybook.js.org/addons/@storybook/addon-controls). Ideally props should be exposed by using this addon, which provides a graphical UI to interact dynamically with the component without needing to write code. Avoid using [Knobs](https://storybook.js.org/addons/@storybook/addon-knobs) for new stories, as this addon is deprecated.
+A great tool to use when writing stories is the [Storybook Controls addon](https://storybook.js.org/addons/@storybook/addon-controls). Ideally props should be exposed by using this addon, which provides a graphical UI to interact dynamically with the component without needing to write code.
```
Comment (tyxla) [inline packages/components/CONTRIBUTING.md]: For historical record purposes we could leave in something like:
```suggestion
A great tool to use when writing stories is the [Storybook Controls addon](https://storybook.js.org/addons/@storybook/addon-controls). Ideally props should be exposed by using this addon, which provides a graphical UI to interact dynamically with the component without needing to write code. Historically, we used [Knobs](https://storybook.js.org/addons/@storybook/addon-knobs), but it was deprecated and later removed in [#47152](https://github.com/WordPress/gutenberg/pull/47152).
```

**Review by tyxla — APPROVED** [review]
Great work @ciampo 🚀 

Tests well for me and code looks great

:shipit:

**Comment by ciampo** [comment]
Thank you! I applied your change, will merge as soon as CI is ✅

**Pull request merged**